UnitystationLauncher/ViewModels/PreferencesPanelViewModel.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using HarfBuzzSharp;
+using ReactiveUI;
+using Serilog;
+using UnitystationLauncher.Models;
+using UnitystationLauncher.Models.ConfigFile;
+using UnitystationLauncher.Services.Interface;
+
+namespace UnitystationLauncher.ViewModels
+{
+    public class PreferencesPanelViewModel : PanelBase
+    {
+        public override string Name => "Preferences";
+
+        public override bool IsEnabled => true;
+
+        private string _installationPath;
+        public string InstallationPath
+        {
+            get => _installationPath;
+            set => this.RaiseAndSetIfChanged(ref _installationPath, value);
+        }
+
+        private bool? _invalidInstallationPath;
+        public bool InvalidInstallationPath
+        {
+            get => _invalidInstallationPath ?? false;
+            set => this.RaiseAndSetIfChanged(ref _invalidInstallationPath, value);
+        }
+
+        private bool? _restartClientPrompt;
+        public bool RestartClientPrompt
+        {
+            get => _restartClientPrompt ?? false;
+            set => this.RaiseAndSetIfChanged(ref _restartClientPrompt, value);
+        }
+
+        private readonly IPreferencesService _preferencesService;
+
+        public PreferencesPanelViewModel(IPreferencesService preferencesService)
+        {
+            _preferencesService = preferencesService;
+
+            _installationPath = _preferencesService.GetPreferences().InstallationPath;
+        }
+
+        public void SetInstallationPath(string path)
+        {
+            if (Directory.GetDirectories(path).Length == 0 && Directory.GetFiles(path).Length == 0 && path.All(char.IsAscii))
```

Comment:
does this run for people who run the launcher for the first time? 

How will it behave for Russian players who open launcher for the first time and their default installations folder has Cyrillic characters?

Reply:
It doesn't, by default when it runs for the first time it will just set it depending on the platform. This will end up being whatever the result of this check plus "/Installations" at the end of the path.

https://github.com/unitystation/stationhub/blob/develop/UnitystationLauncher/Services/EnvironmentService.cs#L32